addons/web_editor/tools.py

```diff
@@ -138,27 +138,43 @@ def get_video_embed_code(video_url):
     return Markup('<iframe class="embed-responsive-item" src="%s" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen="true" frameborder="0"></iframe>') % data['embed_url']
 
 
-def get_video_thumbnail(video_url):
-    """ Computes the valid thumbnail image from given URL
+def get_video_thumbnail_url(video_url):
+    """ Computes the valid thumbnail image URL from given URL
         (or None in case of invalid URL).
     """
     source = get_video_source_data(video_url)
     if source is None:
         return None
 
-    response = None
+    thumbnail_url = None
     platform, video_id = source[:2]
     if platform == 'youtube':
-        response = requests.get(f'https://img.youtube.com/vi/{video_id}/0.jpg', timeout=10)
+        thumbnail_url = f'https://img.youtube.com/vi/{video_id}/0.jpg'
     elif platform == 'vimeo':
-        res = requests.get(f'http://vimeo.com/api/oembed.json?url={video_url}', timeout=10)
+        video_details = f'http://vimeo.com/api/oembed.json?url={video_url}'
+        res = requests.get(video_details, timeout=10)
         if res.ok:
             data = res.json()
-            response = requests.get(data['thumbnail_url'], timeout=10)
+            thumbnail_url = data['thumbnail_url']
     elif platform == 'dailymotion':
-        response = requests.get(f'https://www.dailymotion.com/thumbnail/video/{video_id}', timeout=10)
+        thumbnail_url = f'https://www.dailymotion.com/thumbnail/video/{video_id}'
     elif platform == 'instagram':
-        response = requests.get(f'https://www.instagram.com/p/{video_id}/media/?size=t', timeout=10)
+        thumbnail_url = f'https://www.instagram.com/p/{video_id}/media/?size=t'
+
+    return platform, thumbnail_url
+
+
+def get_video_thumbnail(video_url):
+    """ Computes the valid thumbnail image from given URL
+        (or None in case of invalid URL).
+    """
+    platform, thumbnail_url = get_video_thumbnail_url(video_url)
+    if platform is None or thumbnail_url is None:
+        return None
+
+    response = None
+    if platform in ['youtube', 'dailymotion', 'instagram', 'vimeo']:
+        response = requests.get(thumbnail_url, timeout=10)
```

Comment:
Is that `if` necessary?

Reply:
Yes, because it should not be done for "Youku" for example :slightly_smiling_face: I could have done "if not Youku", but I think it is better to include the working cases and update the list when new cases will be added, than to exclude every other ones (in the end, I only grouped all the if's) :slightly_smiling_face:

Reply:
Oh ok, I was confused by the fact `get_video_thumbnail_url` only handle those 4 ones. Then it means that in the case of other platforms we don't have thumbnails? Would not `if thumbnail_url:` be more straightforward here then? :thinking: That would avoid "duplicating" the list of platforms to consider and actually do what you are meant to do "if we got a thumbnail_url, use it"?

Reply:
Aah indeed, and in the end, the `if` is not even necessary, because the condition just above returns if there is no thumbnail...
My bad I answered too quickly :grimacing: